### DISCUSSION TITLE
Smart Subscriptions Plugin: How do I implement smartSubscription if my PubSub class isn't AsyncIterator based?

### DISCUSSION THREAD

**Discussion by NicolasNewman** [hayes/pothos#643] [Q&A]
I'm currently using Pothos with a NextJS application using GraphQL-Yoga as a server and Apollo for the client. I'm currently able to use `graphql-subscriptions` PubSub class but I'd prefer to use the one built into Yoga. My main roadblock is when I instantiate the SchemaBuilder I'm not sure how to convert:
```ts
smartSubscriptions: {
    ...subscribeOptionsFromIterator((name, { pubSub }) => {
	return pubSub.asyncIterator(name);
    }),
},
```
to an equivalent version with Yoga as their PubSub implementation doesn't use asyncIterators. Yoga's PubSub documentation can be found [here](https://www.the-guild.dev/graphql/yoga-server/docs/features/subscriptions#pubsub) and their implementation can be found [here](https://github.com/dotansimha/graphql-yoga/blob/v2/packages/subscription/src/createPubSub.ts)

I'd greatly appreciate any help / guidance!

**Comment by hayes** [comment]
the yoga pubsub.subscribe method returns an async iterator

```ts
smartSubscriptions: {
    ...subscribeOptionsFromIterator((name, { pubSub }) => {
	return pubSub.subscribe(name);
    }),
},
```

**Reply by NicolasNewman** [reply]
Oh, I guess it's that simple, thank you! I didn't realize Repeater was synonymous with that.

**Reply by NicolasNewman** [reply]
If it requires a second argument of type `number`, do you happen to know what that would correspond to? Yoga's documentation don't mention that for their subscribe method.
